mautic/mautic
# Fix report scheduling

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for features or enhancements / 3.2 for bug fixes 
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:
This PR fix two things.

1, We've had duplicate report scheduling for some reports. I don't know why it happend and when, but I've added delete all report scheduler during the remove https://github.com/mautic/mautic/compare/features...kuzmany:fix-report-scheduling?expand=1#diff-e00f1c22f675d8bdab6109a17a28b8df71c282384da5f5a434e6be41362ccb59R66 

![image](https://user-images.githubusercontent.com/462477/101503764-bfb07680-3972-11eb-965f-6c8eda0837bc.png)

2. Other issue was with different timezone for reschedule. 
If you create scheduling It was scheduled to timezone use in Mautic from Configuration
If report was re-scheduled by command, then timezone was UTC
This PR should fixed it, schedule and re-schedule is still in timezone from Mautic configuration



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Set Europe/Paris timezone in Mautic Configuration
3. Create report and create scheduling every day
4. Check the reports_schedulers table in database
5. Your report is scheduled to next day 23:00:00
6. You can change the date with -1 day directy in database and run `php bin/console mautic:reports:schedule`
7. Report is send and re-schedule to next day to 00:00:00 (we expect 23:00:00) same like when we set it from Mautic
8. This PR fixed it.
